### PR TITLE
remove oc adm privileged policy as it's not needed any more

### PR DESCRIPTION
**Pull request by ceastman-ibm**
[ ] Configuration Infrastructure
[x] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by rvennam** [comment]
@knrc please review this openshift PR

**Comment by knrc** [comment]
@jwendell has offered to review

File: content/en/docs/setup/platform-setup/openshift/index.md [hunk 1]
```diff
@@ -13,15 +13,6 @@ test: no
 
 Follow these instructions to prepare an OpenShift cluster for Istio.
 
-By default, OpenShift doesn't allow containers running with user ID 0.
-You must enable containers running with UID 0 for Istio's service accounts
-by running the command below. Make sure to replace `istio-system` if you are
-deploying Istio in another namespace:
-
-{{< text bash >}}
-$ oc adm policy add-scc-to-group anyuid system:serviceaccounts:istio-system
```
Comment (jwendell) [inline content/en/docs/setup/platform-setup/openshift/index.md:22]: Without this I get:
```
Error creating: pods "istiod-7c9c9d46d4-" is forbidden: unable to validate against any security context constraint: [provider restricted: .spec.securityContext.fsGroup: Invalid value: []int64{1337}: 1337 is not an allowed group spec.containers[0].securityContext.runAsUser: Invalid value: 1337: must be in the ranges: [1000580000, 1000589999]]
```

File: content/en/docs/setup/platform-setup/openshift/index.md [hunk 2]
```diff
@@ -34,22 +25,6 @@ After installation is complete, expose an OpenShift route for the ingress gatewa
 $ oc -n istio-system expose svc/istio-ingressgateway --port=http2
 {{< /text >}}
 
-## Privileged security context constraints for application sidecars
-
-The Istio sidecar injected into each application pod runs with user ID 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following commands. Replace `<target-namespace>` with the appropriate namespace.
-
-{{< text bash >}}
-$ oc adm policy add-scc-to-group privileged system:serviceaccounts:<target-namespace>
-$ oc adm policy add-scc-to-group anyuid system:serviceaccounts:<target-namespace>
```
Comment (jwendell) [inline content/en/docs/setup/platform-setup/openshift/index.md:43]: Without this I get:
```
Error creating: pods "details-v1-79f774bdb9-" is forbidden: unable to validate against any security context constraint: [spec.containers[0].securityContext.runAsUser: Invalid value: 1000: must be in the ranges: [1000600000, 1000609999]]
```

File: content/en/docs/setup/platform-setup/openshift/index.md [hunk 3]
```diff
@@ -34,22 +25,6 @@ After installation is complete, expose an OpenShift route for the ingress gatewa
 $ oc -n istio-system expose svc/istio-ingressgateway --port=http2
 {{< /text >}}
 
-## Privileged security context constraints for application sidecars
-
-The Istio sidecar injected into each application pod runs with user ID 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following commands. Replace `<target-namespace>` with the appropriate namespace.
-
-{{< text bash >}}
-$ oc adm policy add-scc-to-group privileged system:serviceaccounts:<target-namespace>
```
Comment (jwendell) [inline content/en/docs/setup/platform-setup/openshift/index.md:42]: This is ok

**Review by jwendell — CHANGES_REQUESTED** [review]
I tried 1.9.0 on a fresh OCP 4.6 installation, and it failed for me. See the comments.

File: content/en/docs/setup/platform-setup/openshift/index.md (hunk 2 above)
Comment (ceastman-ibm) [reply content/en/docs/setup/platform-setup/openshift/index.md:43]: is there a difference between ocp 4.5 and 4.6? i verified that i dont have the namespaces added to the privileged nor the anyuid groups and my pods are starting without that error you have listed above. i dont have the option yet to upgrade to ocp 4.6.

Comment (ceastman-ibm) [reply content/en/docs/setup/platform-setup/openshift/index.md:43]: so maybe we just need the anyuid group still but the privileged group isn't needed anymore?

Comment (jwendell) [reply content/en/docs/setup/platform-setup/openshift/index.md:43]: yep, that's what I said below.

File: content/en/docs/setup/platform-setup/openshift/index.md [hunk 4]
```diff
@@ -34,19 +34,17 @@ After installation is complete, expose an OpenShift route for the ingress gatewa
 $ oc -n istio-system expose svc/istio-ingressgateway --port=http2
 {{< /text >}}
 
-## Privileged security context constraints for application sidecars
+## Anyuid security context constraints for application sidecars
```
Comment (jwendell) [inline content/en/docs/setup/platform-setup/openshift/index.md:37]: You can completely remove the `Anyuid` from the sentence above.

**Review by jwendell — APPROVED** [review]

**Review by rvennam — APPROVED** [review]

**Comment by craigbox** [comment]
@ceastman-ibm @jwendell please cherrypick to 1.9 if you desire.

**Pull request merged**